components/Header/Header.js

```diff
@@ -1,6 +1,30 @@
 import styled from "styled-components";
 import { useRouter } from "next/router";
 
+export default function Header() {
+  const router = useRouter();
+
+  return (
+    <StyledHomepagediv>
+      <StyledHeader>
+        {router.pathname !== "/" &&
+          router.pathname !== "/spotlight" &&
+          router.pathname !== "/my-library" && (
+            <StyledBackButton onClick={router.back}>
+              <span
+                role="image"
+                aria-label="Arrow emoji indicating the return to the previous page"
+              >
+                ⬅️
+              </span>
+            </StyledBackButton>
+          )}
+        <StyledHeadline>Shelfie</StyledHeadline>
+      </StyledHeader>
+    </StyledHomepagediv>
```

Comment:
- consider to  a more descriptive name for the styled component that represents the main container of the header.
- I guess you could simplify the rendering logic with an array
- Use `role="img"` for accessibility and correct emoji usage

```suggestion
    <StyledContainer>
      <StyledHeader>
        {![ "/", "/spotlight", "/my-library"].includes(router.pathname) && (
            <StyledBackButton onClick={router.back}>
              <span
                role="img"
                aria-label="Arrow emoji indicating the return to the previous page"
              >
                ⬅️
              </span>
            </StyledBackButton>
          )}
        <StyledHeadline>Shelfie</StyledHeadline>
      </StyledHeader>
    </StyledContainer>
```